api/types.proto

```diff
@@ -713,3 +727,31 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+// SecretReference is the linkage between a service and a secret that it uses.
+message SecretReference {
+	// Name is the name of the secret that this references.
+	string name = 1;
+
+	// SecretValueID represents the ID of the specific Secret version that we're
+	// referencing. This identifier exists so that SecretReferences don't leak
+	// any information about the secret contents.
+	string secret_value_id = 2 [(gogoproto.customname) = "SecretValueID"];
```

Comment:
Do we allow secret renames?  Currently in our spike implementation if you try to update a secret with a new name it will reject it because you can't change names.

Also, what about updating the secret version?  This can bump older versions off - should we also warn and require a `--force` if the secret is currently being named?